src/game/gameflow.c

```diff
@@ -606,7 +606,9 @@ static bool GameFlow_LoadScriptLevels(struct json_object_s *obj)
             LOG_ERROR("level %d: 'title' must be a string", level_num);
             return false;
         }
-        cur->level_title = strdup(tmp_s);
+        char *title_check = strdup(tmp_s);
+        Shell_ValidateLevelTitle(title_check);
+        cur->level_title = strdup(title_check);
```

Comment:
We don't free the memory anywhere and we don't check if the allocation succeeded.

Reply:
Let's rebase the branch and use the new `Memory_Dup` utility function. (We don't need to check its results.)

Reply:
We don't need to `strdup` again when assigning to `cur->level_title`. Once is enough.